Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraint.cs

```diff
@@ -328,15 +333,24 @@ protected virtual Vector3 CalculateGoalPosition()
             Bounds trackedHandBounds;
 
             if (trackedController != null &&
-                handBounds.Bounds.TryGetValue(trackedController.ControllerHandedness, out trackedHandBounds))
+                handBounds.LocalBounds.TryGetValue(trackedController.ControllerHandedness, out trackedHandBounds))
             {
                 float distance;
                 Ray ray = CalculateProjectedSafeZoneRay(goalPosition, SolverHandler.TransformTarget, trackedController, safeZone, OffsetBehavior);
                 trackedHandBounds.Expand(safeZoneBuffer);
 
+                MixedRealityPose? palmPose = GetPalmPose(trackedController);
+
+                // We need to transform the ray into hand-space before performing the AABB intersection.
+                ray.origin = Quaternion.Inverse(palmPose.Value.Rotation) * (ray.origin - palmPose.Value.Position);
```

Comment:
Since `palmPose` is nullable, we should probably check for `.HasValue` here before accessing the value, similar to what you already added further down:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/17878a572caf2b69a3b2d0f5eada0173c2a82512/Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraint.cs#L539-L544

Reply:
Whoops. Forgot that. Thanks for catching it